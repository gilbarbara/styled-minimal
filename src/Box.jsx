import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  top,
  width,
  zIndex,
} from 'styled-system';

import { textTransform } from './utils/system';

export const Box = styled.div(
  {
    boxSizing: 'border-box',
  },
  alignContent,
  alignItems,
  alignSelf,
  backgroundImage,
  borders,
  borderRadius,
  bottom,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontFamily,
  fontSize,
  fontStyle,
  fontWeight,
  height,
  justifyContent,
  left,
  lineHeight,
  maxHeight,
  maxWidth,
  minHeight,
  minWidth,
  order,
  position,
  ratio,
  right,
  space,
  textAlign,
  textTransform,
  top,
  width,
  zIndex
);

Box.displayName = 'Box';

Box.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node,
  textTransform: PropTypes.string,
  ...alignContent.propTypes,
  ...alignItems.propTypes,
  ...alignSelf.propTypes,
  ...backgroundImage.propTypes,
  ...borders.propTypes,
  ...borderRadius.propTypes,
  ...bottom.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...flex.propTypes,
  ...flexBasis.propTypes,
  ...flexDirection.propTypes,
  ...flexWrap.propTypes,
  ...fontFamily.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...height.propTypes,
  ...justifyContent.propTypes,
  ...left.propTypes,
  ...maxHeight.propTypes,
  ...maxWidth.propTypes,
  ...minHeight.propTypes,
  ...minWidth.propTypes,
  ...order.propTypes,
  ...position.propTypes,
  ...right.propTypes,
  ...ratio.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...top.propTypes,
  ...width.propTypes,
  ...zIndex.propTypes,
};

export default Box;
