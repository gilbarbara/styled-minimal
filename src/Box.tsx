import * as React from 'react';
import styled from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { BoxProps } from './types';

export const StyledBox = styled('div')<BoxProps>(baseStyles, sx, system);

export const Box = React.forwardRef<HTMLDivElement, BoxProps>((props, ref) => (
  <StyledBox ref={ref} {...props} />
));
