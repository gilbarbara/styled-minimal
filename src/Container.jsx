import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { getTheme, withTheme } from './utils/helpers';
import { ContainerStyles } from './utils/system';
import { textAlignType } from './utils/propTypes';

const verticalPaddingStyles = (props) => {
  const { verticalPadding } = props;
  const { grid, gutter } = getTheme(props);

  if (verticalPadding) {
    return `
      padding-bottom: ${gutter[2]}px;
      padding-top: ${gutter[2]}px;
      
      ${grid.md} {
        padding-bottom: ${gutter[3]}px;
        padding-top: ${gutter[3]}px;
      }
    `;
  }

  return '';
};

const horizontalPadding = (props) => {
  const { grid, gutter } = getTheme(props);

  return `
    padding-left: ${gutter[2]}px;
    padding-right: ${gutter[2]}px;
    
    ${grid.md} {
      padding-left: ${gutter[3]}px;
      padding-right: ${gutter[3]}px;
    }
  `;
};

const layoutProp = props => {
  const { layout } = props;
  const { container } = getTheme(props);

  return container.layout[layout] || '';
};

export const StyledContainer = styled.div`
  ${layoutProp};
  ${ContainerStyles.base};
  ${horizontalPadding};
  ${verticalPaddingStyles};
`;

StyledContainer.propTypes = {
  children: PropTypes.node.isRequired,
  /** use the whole Screen */
  layout: PropTypes.oneOf(['flex', 'fullScreen']),
  textAlign: textAlignType,
  /** add padding top/bottom */
  verticalPadding: PropTypes.bool,
};

StyledContainer.defaultProps = {
  verticalPadding: false,
};

export default withTheme(StyledContainer, 'Container');
