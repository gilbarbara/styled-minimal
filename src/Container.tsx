import * as React from 'react';
import styled, { css } from 'styled-components';

import { responsive, spacer } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { ContainerProps } from './types';

export const StyledContainer = styled('div')<ContainerProps>(
  props => {
    const { ySpacing, fullScreen } = props;
    let ySpacingStyles;
    let fullScreenStyles;

    if (ySpacing) {
      ySpacingStyles = css`
        padding-bottom: ${spacer(3)};
        padding-top: ${spacer(3)};

        ${responsive({
          lg: {
            paddingBottom: spacer(4)(props),
            paddingTop: spacer(4)(props),
          },
        })};
      `;
    }

    if (fullScreen) {
      fullScreenStyles = css`
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        min-height: 100vh;
      `;
    }

    return css`
      ${baseStyles};
      margin-left: auto;
      margin-right: auto;
      padding-left: ${spacer(3)};
      padding-right: ${spacer(3)};
      position: relative;
      width: 100%;
      ${fullScreenStyles};
      ${ySpacingStyles};
      ${responsive({
        lg: {
          paddingLeft: spacer(4)(props),
          paddingRight: spacer(4)(props),
        },
      })};
    `;
  },
  sx,
  system,
);

export const Container = React.forwardRef<HTMLDivElement, ContainerProps>((props, ref) => (
  <StyledContainer ref={ref} {...props} />
));

Container.defaultProps = {
  fullScreen: false,
  ySpacing: false,
};
