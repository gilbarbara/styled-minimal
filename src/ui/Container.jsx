import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { calcUnits, flex, grid, gutter } from '../utils/styles';

const extra = ({ variant }) => (flex[variant] || '');

const Wrapper = styled.div`
  margin-left: auto;
  margin-right: auto;
  max-width: 1440px;
  padding-left: ${calcUnits(`${gutter} / 2`)};
  padding-right: ${calcUnits(`${gutter} / 2`)};
  position: relative;
  width: 100%;
  z-index: 10;
  
 ${extra}
    
  ${grid.md} {
    padding-left: ${gutter};
    padding-right: ${gutter};
  }
    
  ${grid.xl} {
    padding-left: ${calcUnits(`${gutter} * 2`)};
    padding-right: ${calcUnits(`${gutter} * 2`)};
  }
`;

const Container = ({ children, variant }) => (
  <Wrapper variant={variant}>{children}</Wrapper>
);

Container.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Container;
