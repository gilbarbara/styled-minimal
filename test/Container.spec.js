import React from 'react';
import { mount } from 'enzyme';

import Box from '../src/Box';
import Container from '../src/Container';

describe('Container', () => {
  const wrapper = mount(
    <Container verticalPadding>
      <Box>
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts. Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth.
      </Box>;
    </Container>
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
