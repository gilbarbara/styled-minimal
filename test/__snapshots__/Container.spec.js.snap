// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render properly 1`] = `
.c1 {
  box-sizing: border-box;
}

.c0 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  padding-bottom: 16px;
  padding-top: 16px;
}

@media (min-width:768px) {
  .c0 {
    padding-bottom: 32px;
    padding-top: 32px;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding-bottom: 64px;
    padding-top: 64px;
  }
}

@media (min-width:768px) {
  .c0 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<Container
  verticalPadding={true}
>
  <div
    className="c0"
  >
    <Box>
      <div
        className="c1"
      >
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts. Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth.
      </div>
    </Box>
    ;
  </div>
</Container>
`;
